src/Builders/FilterBuilder.php

```diff
@@ -114,6 +114,17 @@ public function whereRegexp($field, $value, $flags = 'ALL')
         return $this;
     }
 
+    public function when($value, callable $callback, callable $default = null)
```

Comment:
Can you describe me a use case of this?

Reply:
It's exactly as the Eloquent `when` method. If the first argument passes as true the closure is executed with the query as parameter.
e.g.
`$query->when(true,function ($query) { $query->where(...); });`